### PR TITLE
Allows for multiple states machines in the same struct

### PR DESCRIPTION
**Pull request by zamith**
Why:

* In certain situations you might want to allow for multiple state
  machines in the same struct.

  See #13

This change addresses the need by:

* Allowing for Fsmx.Struct to be included multiple times per module,
  with a different `state_field`, so it runs independently. When calling
  any of the transition function you can optionally pass it the field
  name, so you can control which one to transition. All of the other
  options, including `transitions` are also independent
* Passing the `state_field` to the callbacks as well, so that you can
  have them be different depending on the which field is being
  transitionned. As of now, the old version is still supported, but it
  is encouraged that you use the new version, which will make it simpler
  to add new fields if you wish.

### CODE REVIEW

**Comment by naps62** [comment]
@zamith thanks. this all looks good. let's just wait a couple of days to see if the people who asked for this on the issue have any input, but other than that it's shippable, it seems

File: lib/fsmx.ex [hunk 1]
```diff
@@ -2,29 +2,39 @@ defmodule Fsmx do
   @moduledoc """
   """
 
+  alias Fsmx.Fsm
+
   @type state_t :: binary() | atom()
 
   @spec transition(struct(), state_t()) :: {:ok, struct} | {:error, any}
```
Comment (gf3) [inline lib/fsmx.ex]: i think these `@spec`s will need to be updated, too

**Review by gf3 — COMMENTED** [review]
for `opts` in `Fsmx.Fsm` we use `:state_field` and in `Fsmx` we use `:field`—should we unify these and use the same atom everywhere?

File: lib/fsmx.ex (hunk 1 above)
Comment (zamith) [reply lib/fsmx.ex]: Yes, missed those.

**Comment by zamith** [comment]
> for `opts` in `Fsmx.Fsm` we use `:state_field` and in `Fsmx` we use `:field`—should we unify these and use the same atom everywhere?

I thought of this, I'm fine with it, if it's less confusing. I even considered not piggy backing on `:state_field` and create something new like `:process_name` or `:name`

**Review by gf3 — APPROVED** [review]
looks awesome! can't wait to put this to use

**Review by naps62 — APPROVED** [review]

**Comment by gf3** [comment]
thank you @zamith and @naps62 :tada:

**Pull request merged**